.github/workflows/telemetrygen.yml

```diff
@@ -17,10 +21,30 @@ jobs:
     if: ${{ github.actor != 'dependabot[bot]' }}
     steps:
       - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version: ~1.20.12
+          cache: false
+      - name: Cache Go
+        id: go-cache
+        timeout-minutes: 5
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/go/bin
+            ~/go/pkg/mod
+          key: go-cache-${{ runner.os }}-${{ hashFiles('**/go.sum') }}
       - name: Set up QEMU
         uses: docker/setup-qemu-action@v3
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v3
+      - name: Build binaries
+        run: |
+          GOOS=linux GOARCH=ppc64le make telemetrygen
+          GOOS=linux GOARCH=arm64 make telemetrygen
+          GOOS=linux GOARCH=amd64 make telemetrygen
+          GOOS=linux GOARCH=s390x make telemetrygen
```

Comment:
I guess we could parallelize this to further speed up things. Should we do it now? Also fine to file an issue to do later

Reply:
The build went from 15 to 2 minutes so by itself this is an improvement. How would you parallelize this?

Reply:
Im ok taking those gains to start haha

Reply:
Fair enough :) I was thinking about doing something like the `cross-compile` jobs, but no need to block this now